### PR TITLE
Upgrade dev dependencies

### PR DESCRIPTION
**Pull request by iamnapo**
- Updated devDeps to latest versions.
- Refactored to match latest xo code-style.
- Refactored to fix breaking changes of ava v3.
- Also test against Node 14. (Only a tiny change in `fs.writeFileSync`).

### CODE REVIEW

**Comment by sindresorhus** [comment]
Sure, go ahead with the refactor.

File: index.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const generateGlobTasks = (patterns, taskOptions) => {
 
 		const ignore = patterns
 			.slice(index)
-			.filter(isNegative)
+			.filter(x => isNegative(x))
```
Comment (sindresorhus) [inline index.js:59]:
```suggestion
			.filter(pattern => isNegative(pattern))
```

**Comment by sindresorhus** [comment]
Thanks :)

File: index.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ const checkCwdOption = (options = {}) => {
 	let stat;
 	try {
 		stat = fs.statSync(options.cwd);
-	} catch (_) {
+	} catch {
```
Comment (andi1984) [inline index.js:28]: @sindresorhus I think this issue slipped through with the removed brackets which fails now in the latest 3.0.2 version. Should I make a PR for fixing it?

**Review by andi1984 — COMMENTED** [review]
IMHO there was a bug slipping through in a try-catch block which breaks now the latest release version 3.0.2.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/globby/pull/167#issuecomment-756388829

**Pull request merged**